authors/apps/authentication/serializers.py

```diff
@@ -1,34 +1,113 @@
+import re
+
 from django.contrib.auth import authenticate
 
+from django.core.validators import RegexValidator
 from rest_framework import serializers
+from rest_framework.validators import UniqueValidator
 
 from .models import User
 
+email_expression = re.compile(
+    r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)")
+at_least_number = re.compile(
+    r"^(?=.*[0-9]).*")
+at_least_uppercase = re.compile(
+    r"^(?=.*[A-Z])(?=.*[a-z])(?!.*\s).*")
+at_least_special_char = re.compile(
+    r"^[a-zA-Z0-9_]*$")
 
 class RegistrationSerializer(serializers.ModelSerializer):
-    """Serializers registration requests and creates a new user."""
-
+    # Username must be longer than 4 characters but shorter than 128 characters. 
+    # Usernamr is a required field
+    username = serializers.CharField(
+        max_length=128,
+        min_length=4,
+        required=True,
+        error_messages={
+            "required"  : "Username is required!"
+        }
+    )
+    # Ensure email is present and has the valid format example@mail.com
+    email = serializers.EmailField(
+        required=True,
+        error_messages={
+            "required" : "Email is required!"
+        }
+    )
     # Ensure passwords are at least 8 characters long, no longer than 128
     # characters, and can not be read by the client.
-    password = serializers.CharField(
+    password = serializers.RegexField(
+        regex="^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$",
         max_length=128,
         min_length=8,
-        write_only=True
+        write_only=True,
+        required=True,
+        error_messages={
```

Comment:
This works fine but having a method validation instead of this might help ensure appropriate error messages are returned. e.g **Password MUST have BOTH letters and numbers** should only be returned if the user submits password that doesn't meet this criteria. Checkout this http://www.django-rest-framework.org/api-guide/serializers/#field-level-validation